README.md

```diff
@@ -0,0 +1,180 @@
+
+A typical Node.js script that compiles IDL using webidl2js looks like the following:
+
+```js
+"use strict";
+const WebIDL2JS = require("webidl2js");
+
+const transformer = new WebIDL2JS({ implSuffix: "-impl" });
+
+transformer.addSource("idl", "impls");
+transformer.generate("wrappers").catch(err => {
+  console.error(err.stack);
+  process.exit(1);
+});
+```
+
+The main module's default export is a class which you can construct with a few options:
+
+- `implSuffix`: a suffix used, if any, to find files within the soruce directory based on the IDL file name.
+- `suppressErrors`: set to true to suppress errors during generation.
+
+The `addSource()` method can then be called multiple times to add directories containing `.idl` IDL files and `.js` implementation class files.
+
+Finally, the `generate()` method will generate corresponding wrapper class files in the given directory.
+
+In this example, a file at `idl/SomeInterface.idl` would generate a new wrapper class at `wrappers/SomeInterface.js`, which would refer to an implementation class at `impls/SomeInterface-impl.js`. (In practice, usually at least two of these directories are the same, making `implSuffix` a useful option.)
+
+The transformer will also generate a file named `utils.js` inside the wrapper class directory, which contains utilities used by all the wrapper class files.
+
+Note that webidl2js works best when there is a single transformer instance that knows about as many files as possible. This allows it to resolve type references, e.g. when one operation has an argument type referring to some other interface.
+
+## Features
+
+webidl2js is implementing an ever-growing subset of the Web IDL specification. So far we have implemented:
+
+- Interface types, including:
+  - Inheritance
+  - Attributes
+  - Operations
+  - Constants
+  - Stringifiers
+  - `iterable<>` declarations
+  - Class strings (with the semantics of [heycam/webidl#357](https://github.com/heycam/webidl/pull/357))
+- Dictionary types
+- Union types
+- Callback function types
+- Nullable types
+- `sequence<>` types
+- `record<>` types
+- `Promise<>` types
+- `FrozenArray<>` types
+- `typedef`s
+- Basic types (via [webidl-conversions](https://github.com/jsdom/webidl-conversions))
+- Mixins, i.e. `implements` (with a [notable bug](https://github.com/jsdom/webidl2js/issues/49))
+- Overload resolution (although [tricky cases are not easy on the implementation class](https://github.com/jsdom/webidl2js/issues/29))
+- Variadic arguments
+- `[Clamp]`
+- `[Constructor]`
+- `[EnforceRange]`
+- `[Exposed]` and `[NoInterfaceObject]` (by exporting metadata on where/whether it is exposed)
+- `[PutForwards]`
+- `[Replaceable]`
+- `[SameObject]` (automatic caching)
+- `[TreatNullAs]`
+- `[Unforgeable]`
+- `[Unscopeable]`
+
+Notable missing features include:
+
+- Legacy platform objects, i.e. those using `getter`/`setter`/`deleter` declarations ([in progress](https://github.com/jsdom/webidl2js/pull/48))
+- Partial interfaces
```

Comment:
We do support partial interfaces:

https://github.com/jsdom/webidl2js/blob/90c96b26273ba13480d7f97f000ada06ff10843c/lib/transformer.js#L128-L140